src/nvim/os/env.c

```diff
@@ -427,6 +427,33 @@ char *vim_getenv(const char *name)
     return xstrdup(kos_env_path);
   }
 
+  // XDG Base Directory expansion.
+  if (strncmp(name, "XDG_", 4) == 0) {
+    const char *home = vim_getenv("HOME");
+    if (!home) {
+        return NULL;
+    }
+
+    const char *home_suffix = "";
+    if (strcmp(name, "XDG_DATA_HOME") == 0) {
+      home_suffix = "/.local/share/";
+    } else if (strcmp(name, "XDG_CONFIG_HOME") == 0) {
+      home_suffix = "/.config/";
+    } else if (strcmp(name, "XDG_CACHE_HOME") == 0) {
+      home_suffix = "/.cache/";
+    } else if (strcmp(name, "XDG_RUNTIME_DIR") == 0) {
+      return xstrdup("/tmp/");  // Arbitrary value, any idea?
+    }
```

Comment:
By the way, where are `XDG_{DATA,CONFIG}_DIRS`? One of the suggestions in my issue was initially populating `&runtimepath` from XDG_CONFIG_DIRS+XDG_CONFIG_HOME, but I think that XDG_DATA_{DIRS,HOME} should also be taken into account (reason: /usr/share/vim/vimfiles is where package managers usually put third-party plugins, thus with new directories /usr/share/nvim/site should be included and this is naturally done using XDG_DATA_DIRS, but not XDG_DATA_HOME, XDG_CONFIG_HOME or XDG_CONFIG_DIRS). This way &runtimepath will look like:

    $XDG_CONFIG_HOME/nvim
    $XDG_DATA_HOME/nvim
    $xdg_config_dir/nvim (repeated for each directory in $XDG_CONFIG_DIRS)
    $xdg_data_dir/nvim/site (repeated for each directory in $XDG_DATA_DIRS)
    $VIMRUNTIME (looks something like /usr/share/nvim/runtime)
    $xdg_data_dir/nvim/site/after (repeated for each directory in $XDG_DATA_DIRS)
    $xdg_config_dir/nvim/after (repeated for each directory in $XDG_CONFIG_DIRS)
    $XDG_CONFIG_HOME/nvim/after

`XDG_DATA_DIRS` defaults to `/usr/local/share/:/usr/share/` according to the specification. This is why initial suggestion is using `site` in runtimepath only: if it was `runtime` then $VIMRUNTIME was naturally included (effectively meaning that this setting will become obsolete which is bad for compatibility reasons).